src/test/ui/async-await/issue-64130-4-async-move.stderr

```diff
@@ -1,10 +1,13 @@
-error: future cannot be sent between threads safely
+error[E0277]: future cannot be sent between threads safely
   --> $DIR/issue-64130-4-async-move.rs:15:17
    |
 LL | pub fn foo() -> impl Future + Send {
    |                 ^^^^^^^^^^^^^^^^^^ future returned by `foo` is not `Send`
    |
    = help: the trait `std::marker::Sync` is not implemented for `(dyn std::any::Any + std::marker::Send + 'static)`
+   = note: required because of the requirements on the impl of `std::marker::Sync` for `std::ptr::Unique<(dyn std::any::Any + std::marker::Send + 'static)>`
+   = note: required because it appears within the type `std::boxed::Box<(dyn std::any::Any + std::marker::Send + 'static)>`
+   = note: required because it appears within the type `Client`
```

Comment:
I personally find these obligation notes quite dense and hard to extract useful, actionable help out of - I think it might be better to continue omitting them when we provide the async-specific note.